doc/validation.md

```diff
@@ -2,11 +2,20 @@
 
 BAPCtools does 3 types of validation, one more than the spec (https://www.kattis.com/problem-package-format/spec/problem_package_format#input-validators):
 
+1. input validation, which validates the `.in` file for every test case, typically for syntax, format, and ranges
+2. answer validation, which validates the `.ans` file for every test case, typically for syntax, format, and ranges, but not for correctness
+3. ouput validation, which checks correctness and running time for the output of every submission.
```

Comment:
add that this is only available/allowed when `validation` is not `default`

Reply:
Could you rephrase? I’m not sure what you mean.

Reply:
Yes, I tried to be precise but maybe this needs to be even better.

Answer validation _always_ happens during `bt validate`; and it _always_ uses the(!) output validator (which must uniquely exist, and _may_ be the default output validator.) If there are answer validators, these are also run during answer validation.  Maybe it’s more stringent to just say “`bt answer`” (when that command exists, maybe in a few days) instead of using the term _answer validation_.

Some sentences that use these terms:
1. Answer validation is a _process_ (namely, the one triggered by `bt validate`). Answer validators are _programs_ (namely, the ones residing in `answer_validators`). 
2. Answer validation involves the answer validators and the output validator. 
3. Output validation involves the output validator.

Output validation happens during `bt run` (and when the problem is _live_).